js/src/dom/manipulator.spec.js

```diff
@@ -0,0 +1,219 @@
+      fixtureEl2.innerHTML = '<div data-test3="js" data-test4="js2" ></div>'
+      fixtureEl3.innerHTML = '<div attri="1" ></div>'
+      const div1 = fixtureEl1.querySelector('div')
+      const div2 = fixtureEl2.querySelector('div')
+      const div3 = fixtureEl3.querySelector('div')
+
+      expect(Manipulator.getDataAttributes(div1)).toEqual({
+        test: 'js',
+        test2: 'js2'
+      })
+
+      expect(Manipulator.getDataAttributes(null)).toEqual({})
+
+      var stub = sinon
+        .stub(Object, 'getOwnPropertyDescriptor')
+        .callsFake(function () {
+          return false
+        })
+
+      expect(Manipulator.getDataAttributes(div2)).toEqual({
+        test3: 'js',
+        test4: 'js2'
+      })
+
+      expect(Manipulator.getDataAttributes(div3)).toEqual({})
+
+      stub.restore()
+    })
+*/
+  })
+
+  describe('removeDataAttribute', () => {
+    it('should throw error if element is null', () => {
+      const incorrect = function() {
+        removeDataAttribute(null, 'key')
+      }
+      expect(incorrect).toThrow();
+    })
+
+    it('should throw error if element is undefined', () => {
+      const incorrect = function() {
+        removeDataAttribute(undefined, 'key')
+      }
+      expect(incorrect).toThrow();
+    })
+
+    it('should remove data attribute', () => {
+      fixtureEl.innerHTML = '<div data-key="value" ></div>'
+      const div = fixtureEl.querySelector('div')
+      Manipulator.removeDataAttribute(div, 'key')
+      expect(div.getAttribute('data-key')).toBeNull()
+    })
+
+    it('should remove data attribute in lower case', () => {
+      fixtureEl.innerHTML = '<div data-testkey="value" ></div>'
+      const div = fixtureEl.querySelector('div')
+      Manipulator.removeDataAttribute(div, 'tEStKeY')
+      expect(div.getAttribute('data-testkey')).toBeNull()
+    })
+  })
+
+  describe('offset', () => {
+    it('should return object with two properties top and left, both numbers', () => {
+      fixtureEl.innerHTML = '<div></div>'
+      const div = fixtureEl.querySelector('div')
+      const offset = Manipulator.offset(div)      
+      expect(offset).toBeDefined()
+      expect(offset.top).toEqual(jasmine.any(Number))
+      expect(offset.left).toEqual(jasmine.any(Number))
+    })
+  })
+
+  describe('position', () => {
+    it('should return object with two properties top and left, both numbers', () => {
+      fixtureEl.innerHTML = '<div></div>'
+      const div = fixtureEl.querySelector('div')
+      const position = Manipulator.position(div)      
+      expect(position).toBeDefined()
+      expect(position.top).toEqual(jasmine.any(Number))
+      expect(position.left).toEqual(jasmine.any(Number))
+    })
+  })
+
+  describe('toggleClass', () => {
+    it('should not error out if element is null', () => {
+      Manipulator.toggleClass(null, 'test')
+    })
+
+    it('should not error out if element is undefined', () => {
```

Comment:
you can merge `should not error out if element is null` with `should not error out if element is undefined`